yhttp/core/middlewares.py

```diff
@@ -0,0 +1,46 @@
+"""
+Middlewares, are common modifiers of HTTP handlers
+They should be the immediate decorators of handler functions
+"""
+
+from functools import wraps
+
+
+def selective_fields(field_name='selected_fields'):
+    """
+    Modifies the response body to contain only the fields requested by the
+    caller in the query parameter.
+
+    Handlers must return :class:`dict` when this decorator is used.
+
+    .. code-block::
+
+       from yhttp.core import json
+
+
+       @app.route()
+       @json
+       @selective_fields(field_name='field_names')
+       def get(req):
+           return {'foo': 'bar'}
+
+    :param field_name: query parameter name that specifies the requested
+                       fields, example: application/json
+    :return: decorator for HTTP handlers
+    """
+    def decorator(handler):
+        @wraps(handler)
+        def wrapper(req, *a, **kw):
+            full_response = handler(req, *a, **kw)
```

Comment:
Check the response type first, and do other things on `list`, `scalars` and etc.